metadata-ingestion/docs/sources/iceberg/iceberg.md

```diff
@@ -1,4 +1,175 @@
+2. Other part consists of parameters, such as `env` or `stateful_ingestion` which are standard DataHub's ingestor
+   configuration parameters and are described in another chapter.
+
+This chapter discusses various examples of setting up connection to a catalog, depending on underlying implementation.
+Iceberg is designed to have catalog and warehouse separated, which is reflected in how we configure it. It is especially
+visible when using Iceberg REST Catalog - which can use many blob storages (AWS S3, Azure Blob Storage, MinIO) as a
+warehouse.
+
+Note that, for advanced users, it is possible to specify custom catalog client implementation via `py-catalog-impl`
+configuration option - refer to `pyiceberg` documentation on details.
+
+### Glue catalog + S3 warehouse
+
+The minimal config for connecting to such
+
+```yaml
+source:
+  type: "iceberg"
+  config:
+    env: dev
+    catalog:
+      my_catalog:
+        type: "glue"
+        s3.region: "us-west-2"
+        region_name: "us-west-2"
+```
+
+Where `us-west-2` is the region from which you want to ingest. Above will work assuming your pod or environment in which
+you run your datahub CLI is already authenticated to AWS and has proper permissions granted (see below). If you need
+to specify secrets directly, then use below as the template:
+
+```yaml
+source:
+  type: "iceberg"
+  config:
+    env: dev
+    catalog:
+      demo:
+        type: "glue"
+        s3.region: "us-west-2"
+        s3.access-key-id: "${AWS_ACCESS_KEY_ID}"
+        s3.secret-access-key: "${AWS_SECRET_ACCESS_KEY}"
+        s3.session-token: "${AWS_SESSION_TOKEN}"
+
+        aws_access_key_id: "${AWS_ACCESS_KEY_ID}"
+        aws_secret_access_key: "${AWS_SECRET_ACCESS_KEY}"
+        aws_session_token: "${AWS_SESSION_TOKEN}"
+        region_name: "us-west-2"
+```
+
+Above example uses references to fill credentials (either from Secrets defined in Managed Ingestion or environmental variables).
+It is possible (but not recommended due to security concerns) to provide those values in plaintext, directly in the recipe.
+
+#### Glue and S3 permissions required
+
+The role used by the ingestor for ingesting metadata from Glue Iceberg Catalog and S3 warehouse is:
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": ["glue:GetDatabases", "glue:GetTables", "glue:GetTable"],
+      "Resource": "*"
+    },
+    {
+      "Effect": "Allow",
+      "Action": ["s3:GetObject", "s3:ListBucket", "s3:GetObjectVersion"],
+      "Resource": [
+        "arn:aws:s3:::<bucket used by the warehouse>",
+        "arn:aws:s3:::<bucket used by the warehouse>/*"
+      ]
+    }
+  ]
+}
+```
+
+### Iceberg REST Catalog + MinIO
+
+Below assumes MinIO defines authentication specified with `s3.*` prefix. Note specification of `s3.endpoint`, assuming
+MinIO listens on port `9000` at `minio-host`. The `uri` parameter points at IRC endpoint (in this case `iceberg-catalog:8181`).
```

Comment:
IRC (Iceberg Rest Catalog)

Reply:
Good point, the acronym needs to be explained. Standard technical way would be to write it:
```
Iceberg REST Catalog (IRC)
```